src/main/java/LogicManager.java

```diff
@@ -1,155 +1,169 @@
 public class LogicManager {
-
-    public static final int MAX_NUM_OF_TASKS = 100;
-    private ListKeeper listKeeper;
+    private final ListKeeper listKeeper;
 
     public LogicManager(ListKeeper listKeeper) {
         this.listKeeper = listKeeper;
     }
 
-    private Command processListCommand(String[] words) {
+    private Command getInvalidCommand() {
+        return new Command(CommandType.INVALID_COMMAND);
+    }
+
+    private Command getListCommand(String[] words) {
         boolean hasUnnecessaryArguments = words.length > 1;
         if (hasUnnecessaryArguments) {
-            return new Command(CommandType.INVALID_COMMAND);
+            return getInvalidCommand();
         }
         return new Command(CommandType.LIST);
     }
 
-    private boolean isIndexValidInt(String taskIndexString) {
+    private boolean canStringBeParsedAsInt(String taskIndexString) {
         try {
             Integer.parseInt(taskIndexString);
             return true;
-        } catch (NumberFormatException error) {
+        } catch (Exception exception) {
             return false;
         }
     }
-    private Command processMarkUnmarkCommand(String[] words) {
+    private boolean isValidArgument(String taskIntegerString) {
+        if (!canStringBeParsedAsInt(taskIntegerString)) {
+            return false;
+        }
+        int taskIndex = Integer.parseInt(taskIntegerString);
+        return this.listKeeper.isValidTaskIndex(taskIndex);
+    }
+    private Command getMarkUnmarkCommand(String[] words) {
         if (words.length != 2) {
-            return new Command(CommandType.INVALID_COMMAND);
+            return getInvalidCommand();
         }
-        if (!isIndexValidInt(words[1])) {
-            return new Command(CommandType.INVALID_INDEX);
+        String taskIndexString = words[1];
+        if (!isValidArgument(taskIndexString)) {
+            return getInvalidCommand();
         }
 
-        // Valid Command
-        int taskIndex = Integer.parseInt(words[1]);
+        int taskIndex = Integer.parseInt(taskIndexString);
         MarkUnmarkArguments markUnmarkArguments = new MarkUnmarkArguments(taskIndex);
         if (words[0].equals("mark")) {
             return new Command(CommandType.MARK, markUnmarkArguments);
         }
         return new Command(CommandType.UNMARK, markUnmarkArguments);
     }
 
-    private Command processToDo(String currentInput) {
+    private Command getToDoCommand(String currentInput) {
         // Extract after _todo_, which is 4 characters
         String taskName = currentInput.substring(4);
```

Comment:
I like that you commented here how you got the 4 characters to avoid magic literal